src/Iteration.cpp

```diff
@@ -259,6 +260,21 @@ Iteration::flushGroupBased(uint64_t i)
     flush();
 }
 
+void
+Iteration::flushVariableBased( uint64_t i )
+{
+    if( !written() )
+    {
+        /* create iteration path */
+        Parameter< Operation::OPEN_PATH > pOpen;
+        pOpen.path = "";
+        IOHandler()->enqueue( IOTask( this, pOpen ) );
+        this->setAttribute( "__step__", i );
```

Comment:
Just double-checking: this attribute is now, by virtue of our backend implementation, an ADIOS variable?
Let's just update the quoted layout in the PR description with the final layout when we merge it now :)

Reply:
Yes, it is now an ADIOS variable. PR https://github.com/openPMD/openPMD-api/pull/949 refines this notion a bit further by distinguishing attributes that change over steps from constant one `bool changesOverSteps = false;`.
If the backend doesn't support this, the attribute should not be written at all, and likewise the frontend should have a fallback logic if the attribute is not found in the backend.